FPIS/Views/Login.cs

```diff
@@ -11,6 +13,69 @@ public Login()
             InitializeComponent();
             Theme.FormInstance = this;
             Theme.Set(Themes.LIGHT);
+            btnLogin.Click += LogUserIn;
+        }
+
+        private void LogUserIn(object? sender, EventArgs e)
+        {
+            string employeeId = txtEmpId.Text;
+            string password = txtPassword.Text;
+            LoginAuth authToken = AuthenticateUser(employeeId, password);
+            switch (authToken)
+            {
+                case LoginAuth.NOUSERNAME:
+                    MaterialMessageBox.Show("Kindly provide your Employee ID", "Login Failed");
+                    txtEmpId.Focus();
+                    break;
+                case LoginAuth.NOPASSWORD:
+                    MaterialMessageBox.Show("Kindly provide your password", "Login Failed");
+                    txtPassword.Focus();
+                    break;
+                case LoginAuth.AUTHFAIL:
+                    MaterialMessageBox.Show("Invalid login credentials", "Login Failed");
+                    // The called function currently only clears the password
+                    ResetFields();
+                    break;
+                case LoginAuth.AUTHPASS:
+                    MaterialMessageBox.Show("Welcome", "Login Passed");
+                    break;
+            }
+        }
+
+        private LoginAuth AuthenticateUser(string employeeId, string password)
+        {
+            // Seperately check the username and passwords to show different error messages
+            bool dataProvided = UserProvidedData(employeeId);
+            if (!dataProvided)
+            {
+                return LoginAuth.NOUSERNAME;
```

Comment:
There is no variable called `NOUSERNAME` as you have changed it to `NO_USERNAME` and the others

Reply:
Hmm... On it